### DISCUSSION TITLE
Can Tilck be used to build other projects on the top of it ?

### DISCUSSION THREAD

**Discussion by vvaltchev** [vvaltchev/tilck#185] [Q&A] [answered]
The question is about whether it makes sense or not to consume Tilck as a component for a larger software project, like a specialized pseudo-Linux distro or as a kernel for small embedded projects.

**Comment by vvaltchev** [comment, marked as answer]
While in principle I'd be very glad to see other people consuming Tilck for building something else on the top of it, the reality is that this project does _not_ have yet enough features for any practical applications. It does not have (yet) any storage or networking support nor any other form of I/O that can be used to communicate with it from the external world (other than a PS/2 keyboard and the support for the primary display, pre-configured by system's firmware). Tilck does not support dynamic linking, glibc and most of the non-trivial syscalls that Linux implements.

Therefore, building something on the top of Tilck would require being an _experienced_ kernel developer willing to spend a _long time_ contributing to Tilck in order to close the gap between what's needed and what's currently available in the kernel. If that's your case, let's talk about it.

For non-kernel developers, "closing the gap" would mean delegating the necessary work to me. However, the time and energy I can afford to spend on this project are very limited at the moment and, when I work on this project, I have a roadmap on my own, which probably won't align with the needs of most 3rd party projects.

**TL;DR;** Unfortunately, no, it can't. Tilck does not have yet enough features to be used for _any_ practical purposes.
